modules/search-pipeline-common/src/main/java/org/opensearch/search/pipeline/common/helpers/SearchRequestMap.java

```diff
@@ -0,0 +1,413 @@
+    }
+
+    /**
+     * Checks if the SearchSourceBuilder is empty.
+     *
+     * @return {@code true} if the SearchSourceBuilder is empty, {@code false} otherwise.
+     */
+    @Override
+    public boolean isEmpty() {
+        return source == null;
+    }
+
+    /**
+     * Checks if the SearchSourceBuilder contains the specified property.
+     *
+     * @param key The property to check for.
+     * @return {@code true} if the SearchSourceBuilder contains the specified property, {@code false} otherwise.
+     */
+    @Override
+    public boolean containsKey(Object key) {
+        return get(key) != null;
+    }
+
+    /**
+     * Checks if the SearchSourceBuilder contains the specified value.
+     *
+     * @param value The value to check for.
+     * @return {@code true} if the SearchSourceBuilder contains the specified value, {@code false} otherwise.
+     * @throws UnsupportedOperationException always, as the method is not supported.
+     */
+    @Override
+    public boolean containsValue(Object value) {
+        throw new UnsupportedOperationException("containsValue" + UNSUPPORTED_OP_ERR);
+    }
+
+    /**
+     * Retrieves the value associated with the specified property from the SearchSourceBuilder.
+     *
+     * @param key The SearchSourceBuilder property whose value is to be retrieved.
+     * @return The value associated with the specified property.
+     * @throws IllegalArgumentException if the property is not a String.
+     * @throws SearchRequestMapProcessingException if the property is not supported.
+     */
+    @Override
+    public Object get(Object key) {
+        if (!(key instanceof String)) {
+            throw new IllegalArgumentException("key must be a String");
+        }
+        // This is the explicit implementation of fetch value from source
+        switch ((String) key) {
+            case "from":
+                return source.from();
+            case "size":
+                return source.size();
+            case "explain":
+                return source.explain();
+            case "version":
+                return source.version();
+            case "seq_no_primary_term":
+                return source.seqNoAndPrimaryTerm();
+            case "track_scores":
+                return source.trackScores();
+            case "track_total_hits":
+                return source.trackTotalHitsUpTo();
+            case "min_score":
+                return source.minScore();
+            case "timeout":
+                return source.timeout();
+            case "terminate_after":
+                return source.terminateAfter();
+            case "sort":
+                return source.sorts();
+            case "docvalue_fields":
+                return source.docValueFields();
+            case "script_fields":
+                return source.scriptFields();
+            case "fetch_fields":
+                return source.fetchFields();
```

Comment:
This one has type `FieldAndFormat`, which is tagged with `@opensearch.internal`. I think we probably shouldn't be returning this.